windows/shared/logging/PlogTraceLogging/include/PlogTraceLogging/PlogTraceLogging.hxx

```diff
@@ -0,0 +1,46 @@
+
+#ifndef PLOG_TRACE_LOGGING_HXX
+#define PLOG_TRACE_LOGGING_HXX
+
+#include <windows.h>
+
+#include <TraceLoggingProvider.h>
+
+#include <plog/Appenders/IAppender.h>
+#include <plog/Log.h>
+
+#include <string>
+
+TRACELOGGING_DECLARE_PROVIDER(g_hplogTraceLoggingProvider);
+
+namespace plog
+{
+
+template <class Formatter>
+class TraceLoggingAppender : public IAppender
+{
+public:
+    TraceLoggingAppender()
+    {
+        TraceLoggingRegister(g_hplogTraceLoggingProvider);
+    }
+
+    ~TraceLoggingAppender()
+    {
+        TraceLoggingUnregister(g_hplogTraceLoggingProvider);
+    }
```

Comment:
This makes the appender a bit inflexible. Why not introduce a member variable which is populated with the ctor, then use that for thr `TraceLoggingRegister/Unregister` calls instead? This way, the appender isn't dependent on a global, and can be configured per-client as needed.

```suggestion
public:
    TraceLoggingAppender(const GUID& traceLoggingProvider) :
        m_traceLoggingProvider(traceLoggingProvider)
    {
        TraceLoggingRegister(m_traceLoggingProvider);
    }

    ~TraceLoggingAppender()
    {
        TraceLoggingUnregister(m_traceLoggingProvider);
    }
    
private:
    GUID m_traceLoggingProvider;
```

Reply:
I don't think TraceLoggingRegister takes a simple GUID, but a TraceLoggingHProvider which would require users of the wrapper to separately define the handle and provide it to this Appender. I think that might make the wrapper of TraceLogging.h kind of pointless then.

Reply:
I switched the code to accept a TraceLoggingHProvider, which enables users the flexibility of defining their own guid so we can have different providers for each main.cxx that wants to use etw logging.